tasks/generate-index.yml

```diff
@@ -24,13 +28,23 @@ spec:
         #! /usr/bin/env bash
         set -xe
 
-        # OPM needs the standard docker config directory structure
-        export DOCKER_CONFIG=/tmp/.docker
-        mkdir $DOCKER_CONFIG
-        cp $(workspaces.credentials.path)/.dockerconfigjson $DOCKER_CONFIG/config.json
+        if [[ "$(workspaces.credentials.bound)" == "true" ]]; then
+          # Setup registry credentials for OPM. Combine the default credentials
+          # with those found in the workspace to maintain access the internal
+          # registry.
+          export DOCKER_CONFIG=/tmp/.docker
+          mkdir $DOCKER_CONFIG
+          jq -s '.[0] * .[1]' \
+            $(workspaces.credentials.path)/.dockerconfigjson \
+            $HOME/.docker/config.json \
+            > $DOCKER_CONFIG/config.json
+        fi
+
+        INDEX_IMG_STREAM=$(echo $(params.from_index) | rev | cut -f1 -d/ | rev)
+        FROM_INDEX="$(params.internal_registry)/$(context.taskRun.namespace)/$INDEX_IMG_STREAM"
 
         opm index add \
-          --from-index "$(params.from_index)" \
+          --from-index $FROM_INDEX \
```

Comment:
What if it's a new operator? We'll get the error, that such an image doesn't exist, right?